ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java

```diff
@@ -73,7 +73,14 @@ public boolean exist(String database, String tableName) {
 
   @Override
   public List<String> listDatabases() {
-    return doAs(() -> unifiedCatalog.listDatabases());
+    return doAs(
+        () ->
+            unifiedCatalog.listDatabases().stream()
+                .filter(
+                    database ->
+                        tableFilterPattern == null
+                            || tableFilterPattern.matcher(database + ".table_name").matches())
```

Comment:
What if the table filter is 'db.table' strict match?

Reply:
Maybe we do need to discuss whether the db filter parameter is still needed. As for me, too many Hive dbs indeed impact the experience.
cc @zhoujinsong @wangtaohz @baiyangtx .